pyvista/examples/downloads.py

```diff
@@ -84,7 +84,8 @@ def delete_downloads():
 
     """
     _check_examples_path()
-    shutil.rmtree(pyvista.EXAMPLES_PATH)
+    if os.path.exists(pyvista.EXAMPLES_PATH):
+        shutil.rmtree(pyvista.EXAMPLES_PATH)
```

Comment:
`shutil` says "_path_ must point to a directory". So this could raise if there's a file where the directory should be (this would be a crazy situation). We _could_ change `os.path.exists` to `os.path.isdir`, but then we wouldn't delete the file if it were there, so we'd get errors down the line anyway. At least this here would be an early place to raise in this esoteric situation. (We _could_ handle the "file is there" case, but I don't think we should.)

Reply:
Sure thing, changed.

Reply:
OK, although my point was that this was good as is :smile:

Reply:
Okay so my colleague told me one thing:

What if we use sub-processes, and one sub-process deletes these files while the second one validates positively its existence, and then tries to remove already removed files?

I should probably add try/except block here to cover this issue (which as it turns out, might've been the root cause of our problem in the first place). Are you okay with that?